### ISSUE TITLE
Hidden actions based on rowData attributes

### ISSUE THREAD

**Issue by GabriMagic** [mbrn/material-table#2483]
The problem I found is to hide an action based on row attribute values, for instance:

I have this actual feature:

```const actions= [
{
  icon: () => { ... },
  onclick: (event: any, rowData: ? | ?[]) => {...},
  hidden: true
]
```

But if I would like to hide an action based on the table row data I couldn't, this would be my expected feature:

```const actions= [
{
  icon: () => { ... },
  onclick: (event: any, rowData: ? | ?[]) => {...},
  hidden: (rowData) => rowData.disable
]
```
As I saw in the documentation, the definition of the "hidden" attribute is a simple boolean, so I can't use the rowData to change the "hidden" property

![image](https://user-images.githubusercontent.com/6409499/93879848-65d5d400-fcd4-11ea-9b77-954675304ad3.png)

**Comment by potchy** [comment]
I came across the same issue today.
Here's workaround: https://github.com/mbrn/material-table/issues/609#issuecomment-525228694

**Comment by GabriMagic** [comment]
Thanks for your answer, I'm trying to use this with Typescript and I need to type all variables, in this case I sholud declare all as "any" that wouldn't compile:

![image](https://user-images.githubusercontent.com/6409499/93939533-d4418300-fd22-11ea-83e4-d2d2d2699750.png)

Using this workaround woulnd't fix my problem, also when I declare without type, and trying to use it I get lot of errors:

This compile:

![image](https://user-images.githubusercontent.com/6409499/93939873-6ea1c680-fd23-11ea-907c-badae4f78c2e.png)

But not working....

![image](https://user-images.githubusercontent.com/6409499/93939828-592c9c80-fd23-11ea-95de-82d28bc80d4b.png)


So in conclusion, this workaround could fail in some many cases, and also is not compilant with Typescript where you should type all variables, I don't know if I got some error in my code testing it, but it is not working for me.

By the way thanks a lot  for your answer.

**Comment by GabriMagic** [comment]
I did the feature on my own :laughing:, I hope @mbrn could approve it :pray::

https://github.com/mbrn/material-table/pull/2485

**Comment by zubinshi** [comment]
Material table already has build in feature, you can achieve this with following code:

 actions={[
          rowData => ({
            icon:  'toggle_on',
            hidden: rowData.disable
          })
        ]}

**Comment by GabriMagic** [comment]
We talk about that over with potchy, but this is not the solution, but a workaround, so you can see the feature I added here: https://github.com/mbrn/material-table/pull/2485 I hope it could be in the next release, but thanks for your comment
